### PR TITLE
Users and Admins can only modify own info

### LINKED ISSUE #38: Admin cannot modify users
As a logged in Admin
      I can modify my account data
      But I cannot modify any other user's account data

### PR DESCRIPTION
**Pull request by erinnachen**
User story:
As a logged in Admin
      * I can modify my account data
      * But I cannot modify any other user's account data

User can edit username, password, first and last names, address
closes #38

### CODE REVIEW

File: spec/features/admin_can_modify_own_info_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      click_link "Logout"
+
+      visit "/login"
+      fill_in "Username", with: "newusername"
+      fill_in "Password", with: "newpassword"
+      click_button "Login"
+      
+      expect(page).to have_content "Logged in as newusername"
+    end
+
```
Comment (ksk5280) [inline spec/features/admin_can_modify_own_info_spec.rb]: Can you add a context for this test above line 35? 
context "admin tries to edit another user's info" do

File: spec/features/user_can_modify_own_info_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      visit "/login"
+      fill_in "Username", with: "newusername"
+      fill_in "Password", with: "newpassword"
+      click_button "Login"
+
+      expect(page).to have_content "Logged in as newusername"
+      expect(page).to have_content "Name: Janice Doer"
+      expect(page).to have_content "Address: Union Station, Denver, CO 80202"
+    end
+
```
Comment (ksk5280) [inline spec/features/user_can_modify_own_info_spec.rb]: Can you add a context above line 37?
context "User tries to visit another user's edit page"

File: spec/features/user_can_modify_own_info_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      fill_in "Password", with: "password"
+      click_button "Login"
+
+      visit "/users/#{user2.id}/edit"
+
+      expect(page).to have_content "The page you were looking for doesn't exist"
+    end
+  end
+
+  context "User not logged in" do
```
Comment (ksk5280) [inline spec/features/user_can_modify_own_info_spec.rb]: On line 51, I think the context should be something like "Non-logged in user tries to visit edit page".
When looking at these tests I had to read through the whole test to understand what was being tested, but I think that can be avoided by making it more obvious with what you write in the context and scenario. I think that the context should describe the action and the scenario should describe the result of what the user sees.

**Comment by erinnachen** [comment]
Thanks @ksk5280, I added the contexts and it should be clearer.

**Pull request merged**